website/pages/api-docs/secret/mongodbatlas/index.mdx

```diff
@@ -0,0 +1,249 @@
+
+### Sample Payload
+
+```json
+{
+  "public_key": "yhltsvan",
+  "private_key": "2c130c23-e6b6-4da8-a93f-a8bf33218830"
+}
+```
+
+### Sample Request
+```bash
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/mongodbatlas/config`
+```
+
+## Create/Update Programmatic API Key role
+Programmatic API Key credential types create a Vault role to generate a Programmatic API Key at
+either the MongoDB Atlas Organization or Project level with the designated role(s) for programmatic access. If a role with the name does not exist, it will be created. If the role exists, it will be updated with the new attributes.
+
+| Method   | Path                         |
+| :--------------------------- | :--------------------- |
+| `POST`   | `/roles/:name`     |
+
+
+## Parameters
+
+`name` `(string <required>)` - Unique identifier name of the role name
+`project_id` `(string <required>)` - Unique identifier for the organization to which the target API Key belongs. Use the /orgs endpoint to retrieve all organizations to which the authenticated user has access.
+`roles` `(list [string] <required>)` - List of roles that the API Key needs to have. If the roles array is provided:
+
+  -> **IMPORTANT:** Provide at least one role. Make sure all roles must be valid for the Organization or Project.
+
+  -> **NOTE:** Include all roles that you want this API Key to have. Any roles not in this array are removed.
+
+  - The Organization roles are:
+    - `ORG_OWNER`
+    - `ORG_MEMBER`
+    - `ORG_GROUP_CREATOR`
+    - `ORG_BILLING_ADMIN`
+    - `ORG_READ_ONLY`
+
+  - The Project roles are:
+    - `GROUP_CHARTS_ADMIN`
+    - `GROUP_CLUSTER_MANAGER`
+    - `GROUP_DATA_ACCESS_ADMIN`
+    - `GROUP_DATA_ACCESS_READ_ONLY`
+    - `GROUP_DATA_ACCESS_READ_WRITE`
+    - `GROUP_OWNER`
+    - `GROUP_READ_ONLY`
+
+
+`ip_addresses` `(list [string] <Optional>)` - IP address to be added to the whitelist for the API key. This field is mutually exclusive with the cidrBlock field.
+`cidr_blocks` `(list [string] <Optional>)` - Whitelist entry in CIDR notation to be added for the API key. This field is mutually exclusive with the ipAddress field.
+
+### Sample Payload
+
+```json
+{
+  "project_id": "5cf5a45a9ccf6400e60981b6",
+  "roles": ["GROUP_CLUSTER_MANAGER"],
+  "cidr_blocks": ["192.168.1.3/32"],
+  "ip_addresses": ["192.168.1.3", "192.168.1.4"]
+}
+```
+
+```bash
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/mongodbatlas/roles/test-programmatic-key
+```
+
+### Sample Response
+```json
+{
+  "project_id": "5cf5a45a9ccf6400e60981b6",
+  "roles": ["GROUP_CLUSTER_MANAGER"],
+  "cidr_blocks": ["192.168.1.3/32"],
+  "ip_addresses": ["192.168.1.3", "192.168.1.4"],
+  "organization_id": "7cf5a45a9ccf6400e60981b7",
+  "ttl": "30m",
+  "max_ttl": "1h"
+}
+
```

Comment:
Would look better when rendered if this line were stripped.

Reply:
fixed